### DISCUSSION TITLE
Start relayer using Go

### DISCUSSION THREAD

**Discussion by ionut-mihalache** [cosmos/community#72] [Validators, Nodes, Relayers, etc...]
Hello.

I am looking for some pointers to how I can start a relayer and interact with it using Go. I have a couple of nodes started using Gaia and now I am trying to create a channel between them. I want to do most of the stuff programmatically (if not all) because it would be easier to integrate with another project later on (also written in Go).
I tried to follow the e2e testsuite and tests for ibc-go [1] but I don't understand how to actually do what I want. The testsuite infrastructure is not what I need.

So I would appreciate if you can point me to some examples of code which initiate nodes, communicate with them, create the relayer, send inter-chain transactions, etc.; all done using Go.

Thank you and sorry if the tag is not appropriate.

[1] - [https://github.com/cosmos/ibc-go/tree/main/e2e](https://github.com/cosmos/ibc-go/tree/main/e2e)

**Comment by adizere** [comment]
Could this be of help? https://github.com/strangelove-ventures/interchaintest/

**Reply by ionut-mihalache** [reply]
Thank you for the response. The repo that you shared is very good, but I can't find a way to do some more than just fund transfers. I want to have more control over transaction sending and receiving, for example to know when/if the transaction has been received by the second chain or, even better, when the sent transaction has been received (or the ack has been received by the chain that initiated the transaction).

Are there any other examples in the repo that I did see? Thanks.

**Reply by reecepbcups** [reply]
Yes, we have that test here for learning IBC, to do exactly this :) - https://github.com/strangelove-ventures/interchaintest/blob/main/examples/ibc/learn_ibc_test.go

**Reply by ionut-mihalache** [reply]
Nice. Thank you. This helps but what about multiple transfers/user using one command? So a SendIBCTransfer function that receives a list of transfers. Is there such a function or using exec might a solution for this?

**Reply by reecepbcups** [reply]
Multiple token transfers is coming in IBC v9 for multiple coins. You may be able to submit multiple IBC transfer messages though in a single Tx? never tried that but assume it would work

read about multi-message txs here: https://docs.junonetwork.io/developer-guides/miscellaneous/multi-message-transaction
